core/src/main/java/com/rudderstack/android/sdk/core/RudderDeviceModeTransformationManager.java

```diff
@@ -60,39 +62,53 @@ public void run() {
                         long deviceModeEventsCount = dbManager.getDeviceModeRecordCount();
                         RudderLogger.logDebug("DeviceModeTransformationManager: DeviceModeTransformationProcessor: fetching device mode events to flush to transformation service");
                         if ((deviceModeSleepCount >= config.getSleepTimeOut() && deviceModeEventsCount > 0) || deviceModeEventsCount >= DMT_BATCH_SIZE) {
+                            int retryCount = 0;
```

Comment:
Make this function short, extract out backoff and other implementations

Reply:
I've extracted out most of the things in separate methods, LMK if further enhancements are required.